core_types/src/entry.rs

```diff
@@ -19,29 +40,33 @@ impl PartialEq for Entry {
 
 impl From<String> for Entry {
     fn from(string: String) -> Self {
-        Entry(string)
+        Entry::from_content(&string)
     }
 }
 
 impl From<Entry> for String {
     fn from(entry: Entry) -> Self {
-        entry.0
+        entry.content()
     }
 }
 
 impl AddressableContent for Entry {
     fn content(&self) -> Content {
-        String::from(self.to_owned())
+        self.to_json()
+            .expect("could not convert Entry to Json Content")
     }
 
     fn from_content(content: &Content) -> Self {
-        Entry::from(content.to_string())
+        Entry::from_json(&content.to_string()).expect("could not convert Json Content to Entry")
```

Comment:
@lucksus yes they must be because there is only one entry, it is `Entry`. everything "entry like" is actually just stuffed into value.

in the future `Entry` will be a trait rather than a struct, at which point this would be more meaningful.